test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/EntityReferenceFunctionalTests.cs

```diff
@@ -38,17 +57,36 @@ public void KeyLookupCannotAppearAfterEntityReference()
             PathFunctionalTestsUtil.RunParseErrorPath("Dogs(1)/$ref(1)", ODataErrorStrings.RequestUriProcessor_SyntaxError);
         }
 
+        [Fact]
+        public void EntityReferenceCannotAppearAfterProperty()
+        {
+            PathFunctionalTestsUtil.RunParseErrorPath("People(1)/SSN/$ref", ODataErrorStrings.RequestUriProcessor_ValueSegmentAfterScalarPropertySegment("SSN", "$ref"));
+        }
+
         [Fact]
         public void CountCannotAppearAfterEntityReferenceCollectionProperties()
         {
             PathFunctionalTestsUtil.RunParseErrorPath("Dogs(1)/MyPeople/$ref/$count", ODataErrorStrings.RequestUriProcessor_MustBeLeafSegment(UriQueryConstants.RefSegment));
         }
 
         [Fact]
-        public void EntityReferenceCannotAppearAfterAnEntitySet()
+        public void EntityReferenceCanAppearAfterAnEntitySet()

```

Comment:
>EntityReferenceCanAppearAfterAnEntitySet [](start = 20, length = 40)

please add EntityReferenceCanAppearAfterACollectionValuedNavigationProperty and EntityReferenceCanAppearAfterASingleValuedNavigationProperty.

Reply:
Done